packages/firestore-types/index.d.ts

```diff
@@ -0,0 +1,1029 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FirebaseApp, FirebaseNamespace } from '@firebase/app-types';
+
+/**
+ * Document data (for use with `DocumentReference.set()`) consists of fields
+ * mapped to values.
+ */
+export type DocumentData = { [field: string]: any };
+
+/**
+ * Update data (for use with `DocumentReference.update()`) consists of field
+ * paths (e.g. 'foo' or 'foo.baz') mapped to values. Fields that contain dots
+ * reference nested fields within the document.
+ */
+export type UpdateData = { [fieldPath: string]: any };
+
+/** Settings used to configure a `Firestore` instance. */
+export interface Settings {
+  /** The hostname to connect to. */
+  host?: string;
+  /** Whether to use SSL when connecting. */
+  ssl?: boolean;
+}
+
+export type LogLevel = 'debug' | 'error' | 'silent';
+
+export function setLogLevel(logLevel: LogLevel): void;
```

Comment:
It feels weird to be exporting a setLogLevel function (there is no "setLogLevel" function anywhere that you can directly import), but I think this whole firestore-types d.ts file is essentially an implementation detail (the "real" exports are in index.ts /index.node.ts) and so I guess this is fine, though looks a bit strange.

Reply:
👍